__init__.py

```diff
@@ -662,6 +663,117 @@ def _get_alarm_matches(
         # No matches found
         return (status[2], None)
 
+    def _fallback_get_alarm_matches(self, utt):
+        # TODO this needs to become a class variable both here and above
+        status = ["All", "Matched", "No Match Found", "User Cancelled", "Next"]
+        return_status = status[2]
+        alarm_list = self.settings["alarm"]
+
+        self.log.debug("utt:%s, alarms:%s" % (utt, self.settings["alarm"]))
+
+        # Lingua-franca workaround-001 - day of week and date confuses LF terribly. 
+        # if a day of the week (monday, friday, etc) is included with a
+        # valid date like "monday april 5th" LF will return bad data so we make 
+        # sure we never include both. "monday april 5th" becomes "april 5th".
```

Comment:
That parser's being rewritten, because there are all sorts of problems like this. I'd suggest adding a TODO for MycroftAI/lingua-franca#96 which *should* make this unnecessary, and provide more robust coverage for the same conditions.

Obviously wouldn't be prudent to offer a timetable on that, as there are multiple design choices up in the air in chat, and several higher priorities for everyone on LF. Still, Jarbas has written a lot of the code. He might be due to push, as well.